### PR TITLE
Minor changes to landing page

### PR DESCRIPTION
**Pull request by adegeo**
## Summary

- Adjust "whats new" to point to .NET 5
- Removed Core from overview title

### CODE REVIEW

File: docs/index.yml [hunk 1]
```diff
@@ -64,10 +64,10 @@ conceptualContent:
...
         - url: core/introduction.md
           itemType: overview
-          text: .NET Core overview
-        - url: core/whats-new/dotnet-core-3-1.md
+          text: .NET overview
+        - url: core/whats-new/dotnet-five.md
           itemType: whats-new
-          text: What's new in .NET Core 3.1
+          text: What's new in .NET 5
         - url: ./core/introduction.md
```
Comment (gewarren) [inline docs/index.yml:71]: Tour of .NET and .NET overview seem to be pointing to the same article.

File: docs/index.yml [hunk 2]
```diff
@@ -64,10 +64,10 @@ conceptualContent:
       links:
         - url: core/introduction.md
           itemType: overview
-          text: .NET Core overview
-        - url: core/whats-new/dotnet-core-3-1.md
+          text: .NET overview
+        - url: core/whats-new/dotnet-five.md
```
Comment (tdykstra) [inline docs/index.yml:68]:
```suggestion
        - url: core/dotnet-five.md
```

**Review by tdykstra — COMMENTED** [review]
LGTM but one URL fix

File: docs/index.yml (hunk 1 above)
Comment (adegeo) [reply docs/index.yml:71]: Yeah, unsure where these would go. I'll leave them the same for now.

Comment (tdykstra) [reply docs/index.yml:71]: I would delete the Tour of .NET.

**Review by tdykstra — APPROVED** [review]
LGTM aside from the issue of having two links pointing to the same article.

**Comment by adegeo** [comment]
Moved these changes to #23229

**Pull request closed without merging**